### PR TITLE
bumps bitcoinjs-lib and adds psbt signature parsing support

### PR DESCRIPTION
**Pull request by humanumbrella**
This PR updates bitcoinjs-lib to version `5.1.10` so we can use Psbt class. Also, we add PSBT signature parsing functionality and unit tests.

Had to update the imports in the utils section given a refactor of bitcoinjs-lib.

We'll need to refactor TransactionBuilder as now the tests give warnings that it will be deprecated in a future version of `bitcoinjs-lib`.

Enable future PSBT work for use in `unchained-wallets` and other applications, e.g. `Caravan`

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -2577,7 +2577,7 @@
     },
     "@types/node": {
       "version": "10.12.18",
-      "resolved": "https://registry.npmjs.org/@types/node/-/node-10.12.18.tgz",
+      "resolved": "https://artifacts.unchained-capital.us/repository/npm-dev-group/@types/node/-/node-10.12.18.tgz",
```
Comment (destrys) [inline package-lock.json:2580]: Need to change this back to npm.

File: src/psbt.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    psbt.data.inputs &&
+    psbt.data.inputs[0]
+  ) ? psbt.data.inputs[0].partialSig : undefined;
+  const numSigners = partialSignatures === undefined ? 0 : partialSignatures.length;
+
+  const signatureSet = {};
+  let pubKey = '';
+  const inputs = psbt.data.inputs;
+  // Find signatures in the PSBT
+  if (numSigners === 2) {
```
Comment (destrys) [inline src/psbt.js:44]: This should be >2, caravan support 3-of-5 etc.

File: src/psbt.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+      for (let j = 0; j < numSigners; j++) {
+        pubKey = toHexString(Array.prototype.slice.call(inputs[i].partialSig[j].pubkey));
+        if (pubKey in signatureSet) {
+          signatureSet[pubKey].push(inputs[i].partialSig[j].signature.toString("hex"));
+        } else {
+          signatureSet[pubKey] = [inputs[i].partialSig[j].signature.toString("hex")];
+        }
+      }
+    }
+  } else if (numSigners === 1) {
```
Comment (destrys) [inline src/psbt.js:56]: Could you refactor this so the numSigners == 1 and numSigner > 1 use the same code?

**Review by destrys — CHANGES_REQUESTED** [review]
Left some comments that need addressing.

File: src/psbt.js (hunk 3 above)
Comment (humanumbrella) [reply src/psbt.js:56]: Yeah -- oops - I refactored after initial solution, but missed this simplification.

**Comment by humanumbrella** [comment]
NOTE: this is a major version bump for bitcoinjs-lib - I've tested locally in caravan and can complete testsuite on both Ledger S and Trezor One, but want to make it known explicitly.

**Comment by waldenraines** [comment]
@destrys other comments?

**Review by destrys — APPROVED** [review]
LGTM!

**Review by waldenraines — APPROVED** [review]
LGTM and everything I tested worked well.

Also this:

gscreenshot_2020-09-08-104435
gscreenshot_2020-09-08-104044

**Pull request merged**